apps/shinkai-desktop/src/components/agent/agent-form.tsx

```diff
@@ -259,33 +270,70 @@ function AgentSideChat({
 
-    if (!chatInboxId) {
-      await createJob({
-        nodeAddress: auth.node_address,
-        token: auth.api_v2_key,
-        llmProvider: agentId,
-        content: message,
-        isHidden: false,
-        chatConfig: {
-          stream: DEFAULT_CHAT_CONFIG.stream,
-          custom_prompt: '',
-          temperature: DEFAULT_CHAT_CONFIG.temperature,
-          top_p: DEFAULT_CHAT_CONFIG.top_p,
-          top_k: DEFAULT_CHAT_CONFIG.top_k,
-          use_tools: hasTools,
-        },
-      });
-    } else {
-      await sendMessageToJob({
-        nodeAddress: auth.node_address,
-        token: auth.api_v2_key,
-        jobId: extractJobIdFromInbox(chatInboxId),
-        message: message,
-        parent: '',
-      });
-    }
+    const messageToSend = message;
+    setMessage(''); // Clear input immediately
+
+    try {
+      if (!chatInboxId) {
+        // Create Job first
+        const createJobResponse = await createJob({
+          nodeAddress: auth.node_address,
+          token: auth.api_v2_key,
+          llmProvider: agentId,
+          content: messageToSend, // Use the captured message
+          isHidden: false,
+          chatConfig: {
+            stream: DEFAULT_CHAT_CONFIG.stream,
+            custom_prompt: '',
+            temperature: DEFAULT_CHAT_CONFIG.temperature,
+            top_p: DEFAULT_CHAT_CONFIG.top_p,
+            top_k: DEFAULT_CHAT_CONFIG.top_k,
+            use_tools: hasTools,
+          },
+        });
+
+        const newJobId = createJobResponse.jobId;
+        const newInboxId = buildInboxIdFromJobId(newJobId);
+        // Set chatInboxId immediately so message list can update optimistically
+        setChatInboxId(newInboxId);
 
-    setMessage('');
+        // Then send the message
+        await sendMessageToJob({
+          nodeAddress: auth.node_address,
+          token: auth.api_v2_key,
+          jobId: newJobId,
+          message: messageToSend, // Use the captured message
+          parent: '',
+        });
+
+        // Invalidate the inboxes query AFTER the first message is sent
+        queryClient.invalidateQueries({
+          queryKey: [
+            FunctionKeyV2.GET_AGENT_INBOXES,
+            {
+              agentId: agentId,
+              nodeAddress: auth?.node_address ?? '',
+            },
+          ],
+        });
+      } else {
+        // Send message to existing job
+        await sendMessageToJob({
+          nodeAddress: auth.node_address,
+          token: auth.api_v2_key,
+          jobId: extractJobIdFromInbox(chatInboxId),
+          message: messageToSend, // Use the captured message
+          parent: '',
+        });
+        // No need to invalidate here
+      }
+    } catch (error) {
+      console.error("Error sending message or creating job:", error);
+      // Optional: Add user feedback (toast notification)
+      setMessage(messageToSend); // Restore message input on error
+    } finally {
+      setMessage('');
+    }
```

Comment:
These changes seem to duplicate functionality we already had in the previous version, was there a particular reason for this approach here?

Reply:
The only thing important there is adding the query invalidate for GET_AGENT_INBOXES. I just removed the duplicate sendMessageToJob

Reply:
moved the query invalidate inside the hook to keep consistency 
```
queryClient.invalidateQueries({
          queryKey: [
            FunctionKeyV2.GET_AGENT_INBOXES,
            {
              agentId: agentId,
              nodeAddress: auth?.node_address ?? '',
            },
          ],
        });
```